docs/src/concepts.md

```diff
@@ -0,0 +1,123 @@
+
+<!-- Additional complexity is added because the "LSP Goto Symbol In Project" is asynchronous and should be refreshed upon the user typing another character for the identifier. In contrast, the "Sublime Text Goto Symbol In Project" has all identifiers greedily indexed and ready to be presented. A problem for this greedy behavior is that at start-up of Sublime Text, "Sublime Text Goto Symbol In Project" does not work right away; running the feature presents the user with nothing, and there is no indication of what the problem is.
+
+In any case, these differences and the fact that Sublime Text as of this writing does not provide a suitable Python API endpoint for this feature makes the "LSP Goto Symbol In Project" rather hard to do. Pull requests for improvement are welcome. -->
+
+## Expand Selection [replacing]
+
+Expand Selection can be accessed by clicking on Selection > Expand Selection. A common key binding for this is <kbd>ctrl</kbd><kbd>shift</kbd><kbd>A</kbd>. A language server may also have this capability and is in a better position to decide what a "smart" Expand Selection should do.
+
+## Hover [replacing]
+
+"Hover" is a general term for an informational popup that appears when you bring your mouse to a word in the file. Sublime Text shows the definition(s) and references of the word that is under your caret.
+
+The LSP package replaces this built-in hover popup with your language server's hover info, if it has the capability. For instance, it may display type information, documentation, and example usage.
+
+## Diagnostics [new]
+
+Diagnostics is a general term for "things that are of interest in the file". It may be syntax errors, warnings from your compiler, or hints about unused variables in a function.
+
+It is incorrect to call this "lint results", because diagnostics encompass more than just lint results.
+
+It is also incorrect to call these "problems", as hints are not really problems.
+
+Sublime Text has no concept of diagnostics (nor lint results or problems), and hence does not provide an API endpoint to push diagnostics to the end-user. This package invented its own diagnostics presentation system.
+
+The SublimeLinter package provides similar functionality.
+
+## Formatting [new]
+
+Formatting is the art of computing a minimal set of white space text replacements. Formatting may be applied manually through a command invocation, or automatically when saving the file. Sublime Text has no concept of Formatting.
```

Comment:
> Formatting is the art of computing a minimal set of white space text replacements.

I'm not sure about that. The formatting is more about formatting the document in a certain way. Whether it's done using minimal set of replacements or by replacing the whole document is not really relevant here, I'd say.

Reply:
Well, it's still supposed to only be whitespace changes.